homeassistant/components/light/decora_wifi.py

```diff
@@ -0,0 +1,227 @@
+            message, title=NOTIFICATION_TITLE, notification_id=NOTIFICATION_ID)
+        return False
+
+    # Gather all the available devices...
+    perms = session.residential_permissions()
+    all_residences = []
+    for permission in perms:
+        for res in session.residences(permission['residentialAccountId']):
+            all_residences.append(res)
+    all_switches = []
+    for residence in all_residences:
+        for switch in session.iot_switches(residence['id']):
+            all_switches.append(switch)
+
+    add_devices(DecoraWifiLight(session, switch) for switch in all_switches)
+
+    # Listen for the stop event and log out.
+    def logout(event):
+        """Log out..."""
+        session.logout()
+
+    hass.bus.listen(EVENT_HOMEASSISTANT_STOP, logout)
+
+
+class DecoraWifiSession:
+    """This class represents an authorized HTTPS session with the LCS API."""
+
+    def __init__(self):
+        """Initialize the session, all content is JSON."""
+        self._session = requests.Session()
+        self._session.headers.update({'Content-Type': 'application/json'})
+        self._user_id = None
+        self._email = None
+        self._password = None
+
+    def call_api(self, api, payload=None, method='get'):
+        """Generic method for calling LCS REST APIs."""
+        # Sanity check parameters first...
+        if method != 'get' and method != 'post' and method != 'put':
+            msg = "Tried DecoraWifiSession.call_api with bad method: %s"
+            raise ValueError(msg % method)
+
+        if self._user_id is None and api != '/Person/login':
+            raise ValueError('Tried an API call without a login.')
+
+        uri = LEVITON_ROOT + api
+
+        if payload is not None:
+            payload_json = json.dumps(payload)
+        else:
+            payload_json = ''
+
+        response = getattr(self._session, method)(uri, data=payload_json)
+
+        # Unauthorized
+        if response.status_code == 401 or response.status_code == 403:
+            # Maybe we got logged out? Let's try logging in.
+            self.login(self._email, self._password)
+            # Retry the request...
+            response = getattr(self._session, method)(uri, data=payload_json)
+
+        if response.status_code != 200 and response.status_code != 204:
+            _LOGGER.error("myLeviton API call (%s) failed: %s, %s",
+                          api, response.status_code, response.body)
+            return None
+
+        return json.loads(response.text)
+
+    def login(self, email, password):
+        """Login to LCS & save the token for future commands."""
+        payload = {
+            'email': email,
+            'password': password,
+            'clientId': 'levdb-echo-proto',  # from myLeviton App
+            'registeredVia': 'myLeviton'     # from myLeviton App
+        }
+
+        login_json = self.call_api('/Person/login', payload, 'post')
+
+        if login_json is None:
+            return None
```

Comment:
Seems like bad practice. See accepted answer here: https://stackoverflow.com/questions/15300550/python-return-return-none-and-no-return-at-all